### PR TITLE
Assignment 7 of Liping

### PR DESCRIPTION
**Pull request by ly2452**
Dear Charles,

This is my assignment 7, please check. Thank you!
Best regards!

Yours,
Liping Yang

### CODE REVIEW

File: Assignment7.Rmd [hunk 1]
```diff
@@ -1,81 +1,104 @@
...
-
+library(ggplot2)
+library(GGally)
+ggplot(data, aes(x=post.test.score)) + geom_histogram() + facet_wrap(~level.up)
+ggplot(data, aes(x=pre.test.score)) + geom_histogram() + facet_wrap(~level.up)
+ggplot(data, aes(x=messages)) + geom_histogram() + facet_wrap(~level.up)
+ggplot(data, aes(x=forum.posts)) + geom_histogram() + facet_wrap(~level.up)
+ggplot(data, aes(x=av.assignment.score)) + geom_histogram() + facet_wrap(~level.up)
+ #Then visualize the relationships between variables
+ggpairs(D1[2:6])
```
Comment (charles-lang) [inline Assignment7.Rmd]: Good work on using `facet`, if you `gather` the data first you can graph everything on one page using `facet`

File: Assignment7.Rmd [hunk 2]
```diff
@@ -1,81 +1,104 @@
...
 #Try to capture an intution about the data and the relationships
-
 ```
-#Classification tree
+According to the figure, first we find that there are some differences between the distributions of every variables and it is related to level up. When it comes to the correlation coefficients, we find that there are positive correlation between the variables. Some of them even has a strong correlation.
```
Comment (charles-lang) [inline Assignment7.Rmd]: Great interpretation 👍

File: Assignment7.Rmd [hunk 3]
```diff
@@ -1,81 +1,104 @@
...
+ #Plot and generate a CP table for your tree 
+printcp(rp)
+post(rp, file = "tree.ps", title = "online_data")
 #Generate a probability value that represents the probability that a student levels up based your classification tree 
-
+D1 <- data.frame(level.up = data$level.up)
 D1$pred <- predict(rp, type = "prob")[,2]#Last class we used type = "class" which predicted the classification for us, this time we are using type = "prob" to see the probability that our classififcation is based on.
 ```
-## Part II
+We use pre.test.score, forum.posts and av.assignment.score as our input variable, since they have weak correlation and the model perfomrs well.
```
Comment (charles-lang) [inline Assignment7.Rmd]: A good rationale

File: Assignment7.Rmd [hunk 4]
```diff
@@ -1,81 +1,104 @@
...
+unlist(slot(performance(pred.detail,"auc"), "y.values"))#Unlist liberates the AUC value from the "performance" object created by ROCR
+ #Now repeat this process, but using the variables you did not use for the previous model and compare the plots & results of your two models. Which one do you think was the better model? Why?
+rp1 <- rpart(level.up~., method = 'class', data=data)
+printcp(rp1)
+D2 <- data.frame(level.up = data$level.up)
+D2$pred <- predict(rp1, type = "prob")[,2]
+pred.detail1 <- prediction(D2$pred, D2$level.up)
+plot(performance(pred.detail1, "tpr", "fpr"))
+abline(0, 1, lty = 2)
+unlist(slot(performance(pred.detail1,"auc"), "y.values"))
```
Comment (charles-lang) [inline Assignment7.Rmd]: Great work!

File: Assignment7.Rmd [hunk 5]
```diff
@@ -1,81 +1,104 @@
...
 ```
-## Part III
+We find that in our model 2, it's auc value equal to 1, which represents that model 2 fit the training data very well and makes no error. We can say model 2 is a better model according to the roc curve. But when it comes to overfitting, model 2 may not be the best model. We should use test data to test whether it will be the best model,
```
Comment (charles-lang) [inline Assignment7.Rmd]: Excellent insight! That is exactly correct

File: Assignment7.Rmd [hunk 6]
```diff
@@ -1,81 +1,104 @@
...
+matrix2 <- as.matrix(table2)
+table2
+D1$accuracy.model2 <- 600/nrow(D1)
+D1$precision.model2 <- 600/(600+0)
+D1$recall.model2 <- 600/(600+400)
+ kappa(matrix2, exact = TRUE)/kappa(matrix2)
 ```
-
-
+From the three diagnostics, we can see that threshold 1 is better to predict level.up. And acutually if we look at the roc curve, we will also find that threshold 1 is the best threshold. It not only has high accuracy and precision but also has high recall value. Futher more, threshold 1 has a higher kappa value than threshold 2, which shows that threshold 1 is better.
```
Comment (charles-lang) [inline Assignment7.Rmd]: Do you have any thoughts about why that might be the case? Can you interpret this in terms of overfitting.
